### PR TITLE
Add yamato configurations for building and uploading artifact to Stevedore.

### PR DESCRIPTION
**Pull request by brian-taylor-unity**
Added yamato configurations for uploading runtime artifacts to stevedore for Windows x86, Windows x64, OSX x64, and Linux x64.

I separated the build+test yamato configs into separate build and test configs, where the test and publish-to-stevedore configs are dependent on the build configs.

Build configs will generate artifacts for both the full directory structure and for just the runtime 7z we use for stevedore. I have the build configs generate the stevedore artifacts instead of the publish-to-stevedore configs because a developer might want to get a built .7z for local testing before actually publishing that build to stevedore.

Test configs will still trigger on any PR targeting trunk, but publish-to-stevedore configs are all manual. There is a top-level "Publish to Stevedore" config that will trigger all of the platforms' publish-to-stevedore configs.

Where we publish the artifacts to (`testing`, `unity-internal`, or `public`) is set and can be changed in `.yamato\Globals.metafile`.

### CODE REVIEW

**Comment by brian-taylor-unity** [comment]
I'll need to change the Github PR checks after merging this to reflect the new names of yamato configs.

**Review by UnityAlex — APPROVED** [review]
LGTM assuming all the checks pass

**Review by joshpeterson — APPROVED** [review]

**Pull request merged**
